lib/imgix/path.rb

```diff
@@ -129,6 +127,39 @@ def method_missing(method, *args, &block)
 
     private
 
+    # Escape and encode any characters in path that are reserved and not utf8 encoded.
+    # This includes " +?:#" characters. If a path is being used as a proxy, utf8
+    # encode everything. If it is not being used as proxy, leave certain chars, like
+    # "/", alone. Method assumes path is not already encoded.
+    def sanitize_path(path)
+      # remove the leading "/", we'll add it back after encoding
+      path = path.slice(1, path.length) if Regexp.new('^/') =~ path
+      # if path is being used as a proxy, encode the entire thing
+      if /^https?/ =~ path
+        return encode_URI_Component(path)
+      else
+        # otherwise, encode only specific characters
+        return encode_URI(path)
+      end
+    end
```

Comment:
In the Variant PR, which shifts a lot of this escaping/encoding work to application boot, this would ideally be done in the Variant initializer: https://github.com/imgix/imgix-rb/pull/104/files#diff-f7f33d40dff4d8a40100ac8a68f7fab2e082a98a683b678c8ee04c1bc930102aR8

Reply:
Haven't had a chance to dig into #104 too much, but loved what I saw so far. Will definitely keep this in mind.

Reply:
Thanks for the suggestion @stevehodgkiss. I could not find any encoding issues when `to_url` is called multiple times on the same `Path`, but I’m sure there will be edge cases that I haven’t thought of. Also, I think the ivar for `sanitized_path` creates a more predictable experience for the user.

I’ve implemented a solution based on your idea:
https://github.com/imgix/imgix-rb/blob/bd5bdecac452a676e7f87d8b6b5e4f86485a9523/lib/imgix/path.rb#L17-L27

Before these changes, `@path` was being modified in a way that was not explicit. There may be a case in the future where we want an `@sanitized_path` var or a `santize_path!` func, but for the purposes of this PR that is out of scope.

There are a lot of improvements needed in this file and the library throughout, but I felt these changes have to least potential for producing breaking changes.

Reply:
Looks good! I take back my previous comment about "this would ideally be done in the Variant initializer". This can't be done there since it's about the Path itself, not the Variant.